data-prepper-core/src/test/java/org/opensearch/dataprepper/event/DefaultEventFactoryTests.java

```diff
@@ -16,46 +17,56 @@
 import org.apache.commons.lang3.RandomStringUtils;
 
 import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
 import java.util.Collections;
 
 class DefaultEventFactoryTests {
-    private DefaultEventBuilder eventBuilder;
-    private DefaultLogEventBuilder logEventBuilder;
     private DefaultEventFactory eventFactory;
     
+    public DefaultEventFactory createObjectUnderTest() {
+        List<DefaultEventBuilderFactory> factories = new ArrayList<>();
+        factories.add(new DefaultEventBuilderFactory());
```

Comment:
You can must mock these rather than rely on the existing objects.

```
Class class1 = mock(Class.class); // I'm not sure if you can mock this. Maybe use an arbitrary class if that doesn't work.
EventBuilderFactory factory1 = mock(EventBuilderFactory.class);
when(factory1.getClass).thenReturn(class1);
DefaultBaseEventBuilder builder1 = mock(DefaultBaseEventBuilder.class);
when(factory1.createNew()).thenReturn(builder1);
```

In the tests, you can write something like:

```
assertThat(createObjectUnderTest(). eventBuilder(class1), equalTo(builder1);
assertThat(createObjectUnderTest(). eventBuilder(class2), equalTo(builder2);
```